File: 1-two-sum/two-sum.c
// Problem: https://leetcode.cn/problems/two-sum/
// Accepted at: 2024年4月27日 11:43

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int* result = (int*)malloc(sizeof(int) * 2);
    int i, j;
    for (i = 0; i < numsSize - 1; i++) {
        for (j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                *returnSize = 2;
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
    }
    *returnSize = 0;
    return NULL;
}

File: 1-two-sum/two-sum.js
// Problem: https://leetcode.cn/problems/two-sum/
// Accepted at: 2025年9月23日 09:13

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const m = new Map();
    let result = []
    for(let [index, n] of nums.entries()){
        const diff = target - n;
        if(m.has(diff)){
            result = [m.get(diff), index];
        } else {
            m.set(n, index);
        }
    }
    return result;
};
